kaldb/src/main/java/com/slack/kaldb/writer/kafka/KaldbKafkaConsumer.java

```diff
@@ -275,43 +275,42 @@ public boolean consumeMessagesBetweenOffsetsInParallel(
       ConsumerRecords<String, byte[]> records =
           kafkaConsumer.poll(Duration.ofMillis(kafkaPollTimeoutMs));
       int recordCount = records.count();
-      LOG.debug("Fetched records={} from partition:{}", recordCount, topicPartition.partition());
+      LOG.debug("Fetched records={} from partition:{}", recordCount, topicPartition);
       if (recordCount > 0) {
         messagesIndexed += recordCount;
         executor.execute(
             () -> {
-              LOG.info("Ingesting batch: [{}/{}]", topicPartition.partition(), recordCount);
+              LOG.info("Ingesting batch: [{}/{}]", topicPartition, recordCount);
               for (ConsumerRecord<String, byte[]> record : records) {
                 if (startOffsetInclusive >= 0 && record.offset() < startOffsetInclusive) {
-                  throw new IllegalArgumentException(
-                      "Record is before start offset range: " + startOffsetInclusive);
-                }
-                if (endOffsetInclusive >= 0 && record.offset() > endOffsetInclusive) {
-                  throw new IllegalArgumentException(
-                      "Record is after end offset range: " + endOffsetInclusive);
-                }
-                try {
-                  if (logMessageWriterImpl.insertRecord(record)) {
-                    recordsReceivedCounter.increment();
-                  } else {
-                    recordsFailedCounter.increment();
+                  LOG.warn("Record is before start offset range: " + startOffsetInclusive);
```

Comment:
Couple things on these log messages - the first is they should be using the token replacement instead of string concat. Additionally, since these now skip the record I would make that clearer in the message (ie, "record is before start, will permanently drop this record").

I would also suggest making this a _debug_ instead of a _warn_, or rethink where this log message will sit. As it is currently this has the chance to easily generate millions or billions of log messages in a short amount of time depending on how far a cluster is behind. 

Another way to address this would be to check the first or last record of a batch and only log it once per batch.

Reply:
Good call